src/gpg2john.c

```diff
@@ -1626,10 +1694,13 @@ parse_packet(char *hash)
 		}
 		// else printf("(%d bytes)\n", len);
 
-		if (tag < TAG_NUM && tag_func[tag] != NULL) {
+		if (tag < TAG_NUM && (tag_func[tag] != NULL || tag_func4[tag] != NULL)) {
 			if (gpg_dbg)
 				fprintf(stderr, "Packet type %d, len %d at offset %d  (Processing) (pkt-type %s) (Partial %s)\n", tag, len, offset, pkt_type(tag), partial?"yes":"no");
-			(*tag_func[tag])(len, 1, partial, hash);	// first packet (possibly only one if partial is false).
+			if (tag_func[tag] != NULL)
+				(*tag_func[tag])(len);
+			else
+				(*tag_func4[tag])(len, 1, partial, hash);	// first packet (possibly only one if partial is false).
```

Comment:
None of the `john-samples` examples are calling this else part. Anyway, the call to the method is correct.